ludwig/features/category_feature.py

```diff
@@ -137,13 +141,26 @@ def get_feature_meta(column, preprocessing_parameters, backend):
 
     @staticmethod
     def feature_data(column, metadata):
-        return column.map(
-            lambda x: (
-                metadata["str2idx"][x.strip()]
-                if x.strip() in metadata["str2idx"]
-                else metadata["str2idx"][UNKNOWN_SYMBOL]
+        def __replace_token_with_idx(value: Any, metadata: Dict[str, Any], unknown_symbol_idx: int) -> int:
+            stripped_value = value.strip()
+            if stripped_value in metadata["str2idx"]:
+                return metadata["str2idx"][stripped_value]
+            logger.warning("Using unknown symbol during category feature processing.")
```

Comment:
Let's add more to this warning.

> Encountered unknown symbol '{stripped_value}' for '{column.name}' during category feature preprocessing. This should never happen during training. If this happens during inference, this may be an indication that not all possible symbols are present in your training set. Consider re-splitting your data to ensure full representation, or setting `preprocessing.most_common` to be smaller than this feature's total vocabulary size, {len(metadata["str2idx"])}, which will ensure that the model is architected and trained with an `UNKNOWN` symbol.

Reply:
@justinxzhao Sounds good! Just to confirm, we're thinking of leaving this as a warning and not a Exception right? Should I still replace it with a default value like the mode, or maybe something even simpler like the first token?

Reply:
Warning SGTM since otherwise we'd need to change all of the tests using synthetic datasets with rows=25/vocab_size=10.

Updated warning:

> Encountered unseen symbol '{stripped_value}' for '{column.name}' during category feature preprocessing. This should never happen during training. If this happens during inference, this may be an indication that not all possible symbols are present in your training set. Consider re-splitting your data to ensure full representation, or setting preprocessing.most_common to be smaller than this feature's total vocabulary size, {len(metadata["str2idx"])}, which will ensure that the model is architected and trained with an explicit UNKNOWN symbol. For now, examples using '{stripped_value}' will be mapped to `{metadata["idx2str"][fallback_symbol_idx]}`.